NModbus/Device/ModbusMasterTcpConnection.cs

```diff
@@ -16,21 +16,27 @@ namespace NModbus.Device
     /// <summary>
     /// Represents an incoming connection from a Modbus master. Contains the slave's logic to process the connection.
     /// </summary>
-    internal class ModbusMasterTcpConnection : ModbusDevice, IDisposable
+    public class ModbusMasterTcpConnection : ModbusDevice, IDisposable
     {
         
-        private readonly TcpClient _client;
-        private readonly string _endPoint;
-        private readonly Stream _stream;
-        private readonly IModbusSlaveNetwork _slaveNetwork;
-        private readonly IModbusFactory _modbusFactory;
-        private readonly Task _requestHandlerTask;
+        protected internal TcpClient _client;

```

Comment:
You are right, at first I was hoping to change it to "protected readonly" so that subclasses could inherit, but apparently I forgot to add "readonly"，